### PR TITLE
Coinmarketcap historical endpoint

### PR DESCRIPTION
**Pull request by bogos**
## Description

......

## Changes

- Add coinmarketcap historical endpoint

## Steps to Test

1. yarn test coinmarketcap

## Quality Assurance

- [X ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [ ] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [ ] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [ X] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

File: packages/sources/coinmarketcap/src/endpoint/historical.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  const symbol = validator.validated.data.symbol?.toUpperCase() || 'BTC'
+  const time_start = validator.validated.data.start
+  const time_end = validator.validated.data.end
+  const count = validator.validated.data.count || 10
+  const interval = validator.validated.data.interval || '5m'
+  const convert = validator.validated.data.convert?.toUpperCase() || 'USD'
+  const convert_id = validator.validated.data.cid
+
+  const aux = validator.validated.data.aux
+  const skip_invalid = validator.validated.data.skipInvalid || true
```
Comment (austinborn) [inline packages/sources/coinmarketcap/src/endpoint/historical.ts:33]: Will this... work as intended? Assuming `validator.validated.data.skipInvalid` is a boolean or undefined, `validator.validated.data.skipInvalid === false` will return `true` for `skip_invalid`. Alternative would be `?? true`, which returns `true` when the lefthand side is `null` or `undefined`.

**Review by austinborn — CHANGES_REQUESTED** [review]

**Review by austinborn — DISMISSED** [review]

File: packages/sources/coinmarketcap/README.md [hunk 2]
```diff
@@ -392,3 +392,89 @@ https://pro-api.coinmarketcap.com/v1/cryptocurrency/quotes/latest
   "statusCode": 200
 }
 ```
+
+## Historical Endpoint
+
+https://pro-api.coinmarketcap.com/v1/cryptocurrency/quotes/historical33333
```
Comment (boxhock) [inline packages/sources/coinmarketcap/README.md:398]: I'm guessing these 3's are not supposed to be here? :sweat_smile: 

```suggestion
https://pro-api.coinmarketcap.com/v1/cryptocurrency/quotes/historical
```

**Review by boxhock — CHANGES_REQUESTED** [review]
Just one minor issue, otherwise looks good

File: packages/sources/coinmarketcap/README.md [hunk 3]
```diff
@@ -392,3 +392,89 @@ https://pro-api.coinmarketcap.com/v1/cryptocurrency/quotes/latest
...
+## Historical Endpoint
+
+https://pro-api.coinmarketcap.com/v1/cryptocurrency/quotes/historical
+
+### Input Params
+
+| Required? |                  Name                   |                                   Description                                   |                                          Options                                          | Defaults to |
+| :-------: | :-------------------------------------: | :-----------------------------------------------------------------------------: | :---------------------------------------------------------------------------------------: | :---------: |
+|    ✅     | `base`, `from`, `coin`, `symbol`, `sym` |                       The symbol of the currency to query                       |                                                                                           |    `BTC`    |
+|           |   `quote`, `to`, `market`, `convert`    |                    The symbol of the currency to convert to                     |                                                                                           |    `USD`    |
```
Comment (boxhock) [inline packages/sources/coinmarketcap/README.md:405]: It seems like this input param is not defaulting to "USD", and should be required

**Review by boxhock — CHANGES_REQUESTED** [review]
Just noticed this, should be good to merge after the README is updated

File: packages/sources/coinmarketcap/README.md (hunk 3 above)
Comment (boxhock) [reply packages/sources/coinmarketcap/README.md:405]: Same goes for the above, should not be default "BTC"

**Review by boxhock — APPROVED** [review]

**Pull request merged**